packages/cli/src/util/input/edit-project-settings.ts

```diff
@@ -2,45 +2,110 @@ import inquirer from 'inquirer';
 import confirm from './confirm';
 import chalk from 'chalk';
 import { Output } from '../output';
-import { Framework } from '@vercel/frameworks';
+import { Framework, frameworks } from '@vercel/frameworks';
 import { isSettingValue } from '../is-setting-value';
 import { ProjectSettings } from '../../types';
+import { VercelConfig } from '../dev/types';
 
-export interface PartialProjectSettings {
-  buildCommand: string | null;
-  outputDirectory: string | null;
-  devCommand: string | null;
-}
+const settingKeys = [
+  'buildCommand',
+  'installCommand',
+  'devCommand',
+  'ignoreCommand',
+  'outputDirectory',
+  'framework',
+] as const;
```

Comment:
Can we avoid defining these twice?

How about `const settingKeys = Object.keys(settingMap)` instead?

Reply:
Unfortunately, `Object.keys` doesn't maintaint the type automatically so I will have to write them twice regardless... but as:

```ts
const settingKeys = Object.keys(settingMap) as unknown as readonly ["buildCommand", "installCommand", "devCommand", "ignoreCommand", "outputDirectory", "framework"];
```

Reply:
@Ethan-Arrowood This works:

```ts
const settingKeys = Object.keys(settingMap);

type ConfigKeys = keyof typeof settingMap;
```

Reply:
And you don't need `settingsKeys` at all because its just used for iteration and you can iterate over an object

Reply:
Yes but a lot of the code relies on `settingKeys` being of type `readonly ["buildCommand", "installCommand", "devCommand", "ignoreCommand", "outputDirectory", "framework"]` (essentially a tuple of those strings)

Reply:
I can do this tho: 

```ts
type ConfigKeys = keyof typeof settingMap;
const settingKeys = Object.keys(settingMap) as unknown as readonly [ConfigKeys];
```